app/models/skin.rb

```diff
@@ -188,6 +188,7 @@ def cache!
     self.public = true
     self.official = true
     save!
+    Rails.cache.increment('skins_generation')
     css_to_cache = ""
```

Comment:
Its because skins have a parent skin id and the if the parent changes then all the children have to change.

I think it would make sense to add the type of skin to the key maybe eg a work skin or a site skin ?